libs/market-list/src/lib/utils/market-list.utils.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  MarketList,
+  MarketList_markets,
+  MarketList_markets_candles,
+} from '../components/__generated__/MarketList';
+
+export const priceChange = (m: MarketList_markets) => {
+  if (
+    m.candles &&
+    m.candles.length > 0 &&
+    m.candles[m.candles.length - 1]?.open &&
+    m.candles[m.candles.length - 1]?.close
+  ) {
+    const yesterdayLastPrice = Number(m.candles[0]?.close);
+    const recentLastPrice = Number(m.candles[m.candles.length - 1]?.close);
+    const increase = recentLastPrice - yesterdayLastPrice;
+
+    return (increase / yesterdayLastPrice) * 100;
+  }
+  return 0;
+};
+
+export const candles = (m: MarketList_markets) =>
+  m.candles?.map((c: MarketList_markets_candles | null) => ({
+    open: price(c?.open, m.decimalPlaces),
```

Comment:
Chore (blocking): no need for the elvis operator on `c`, as per the comment above, we will only hit this code if `c` is non-nullish. Also chore (non-blocking): destructuring preferred :)